wee_slack.py

```diff
@@ -3311,6 +3314,39 @@ def linkify_word(match):
     return re.sub(linkify_regex, linkify_word, message_escaped, re.UNICODE)
 
 
+def unfurl_blocks(message_json):
+    block_text = []
+    for block in message_json["blocks"]:
+        try:
+            if block["type"] == "section":
+                if "text" in block: block_text += unfurl_texts([block["text"]])
+                if "fields" in block: block_text += unfurl_texts(block["fields"])
+            elif block["type"] == "actions":
+                block_text.append("|".join(i["text"]["text"] for i in block["elements"]))
+            elif block["type"] == "call":
+                block_text.append(". Join via ").append(block["call"]["v1"]["join_url"])
+            elif block["type"] == "divider":
+                block_text.append("\n")
+            elif block["type"] == "context":
+                block_text.append("|".join(i["text"] for i in block["elements"]))
+            else:
+                raise NotImplementedError("Block type not implemented", block["type"])
+        except Exception as e:
```

Comment:
What exceptions can be end up here? I don't it's a good idea to catch general exceptions as you may hide bugs in your implementation.

Reply:
Thanks for reviewing this!  The problem here is that if there is an error in my implementation, it will just not display the message, and you may or may not see the error (I think it goes in the weechat buffer).  This way it will at least display the message and then log the error.  It may be less of a concern now that my implementation has somewhat fewer errors, but it was really annoying to not have messages show up.  What do you think the best solution is here?

Reply:
Right, I forgot that because of the exception handling in `handle_next` errors are just silently ignored when you have debug_mode off. I've been meaning to change that.

Okay, we can keep the exception handler. Though, at some point, I'll probably change the exception handler in `handle_next` to print an error saying you should report the issue to github when exceptions occur. If that is after we have after verified that it matches Slacks documentation for blocks I think we can remove the exception handler here.

I can't find any documentation for the `call` type though. Do you, or @gianarb know if that exists? I assume the implementation here just matches what Zoom sends?

Reply:
Yeah, I think the ideal solution would be to render as much as we can (not sure how that would work), and then add a message (perhaps in message_deleted color) to see more in the debug log.

I think slack's documentation is really lacking.  I haven't seen anything about call blocks.